src/middleware/tokenHandler.js

```diff
@@ -68,6 +68,9 @@ module.exports = (router) => {
       req.user = user;
 
       // Store the jwt token sent by the user.
+      if (decoded.user._id === 'external') {
+        decoded.user._id = decoded.user.data.id || decoded.user.data._id ||decoded.user.data.email || JSON.stringify(decoded.user.data);
```

Comment:
I would prefer that this create a valid "id" hash from the email address. Something like the following 

https://github.com/formio/formio-server/blob/fbde6147e1bf0fdbe0113fb6113e0902e6a9eaf1/src/saml/saml.js#L145
https://github.com/formio/formio-server/blob/fbde6147e1bf0fdbe0113fb6113e0902e6a9eaf1/src/util/util.js#L20